pkg/jx/cmd/clients/factory.go

```diff
@@ -382,13 +383,45 @@ func (f *factory) ResetSecretsLocation() {
 
 // CreateSystemVaultClient gets the system vault client for managing the secrets
 func (f *factory) CreateSystemVaultClient(namespace string) (vault.Client, error) {
-	name, err := kubevault.SystemVaultName(f.kubeConfig)
+	name, err := f.getVaultName(namespace)
 	if err != nil {
-		return nil, errors.Wrap(err, "building the system vault name from cluster name")
+		return nil, err
 	}
 	return f.CreateVaultClient(name, namespace)
 }
 
+func (f *factory) getVaultName(namespace string) (string, error) {
+	name := ""
+	context, err := cluster.Context(f.kubeConfig)
+	if err != nil {
+		return name, err
+	}
+	if context == nil {
+		kubeClient, _, err := f.CreateKubeClient()
```

Comment:
Is it the client created successfully if the context is nil?

Reply:
if there's no config then we can still create a k8s client yeah: https://github.com/jenkins-x/jx/blob/976b6e8d676ed1c6bfda05db85b3727876769c8a/pkg/jx/cmd/clients/factory.go#L617